src/pages/MerchDetail/MerchDetail.tsx

```diff
@@ -73,25 +75,36 @@ export const MerchDetail: React.FC = () => {
     const value = parseInt(target.value, 10);
     if (value <= 0) {
       setQuantity(1);
-    } else if (value > 99) {
-      setQuantity(99);
+    } else if (value > maxQuantity) { 
+      setQuantity(maxQuantity);
     } else {
       setQuantity(value);
     }
   };
 
+  const updateMaxQuantity = (colorway: string, size: string) => {
+    if (product) {
+      const stockQty = getQtyInStock(product, colorway, size);
+      const cartQty = getQtyInCart(cartState.items, product.id, colorway, size);
+      const max = (stockQty > cartQty) ? stockQty - cartQty : 0;
```

Comment:
currently setting the maximum quantity to be the quantity in stock - the quantity already added to cart. 
@chanbakjsd should we consider preventing someone from buying the entire stock?

Reply:
If 2 people attempt to do this at once, the backend should handle this correctly and error as input validation should fail as a result. However, frontend will need to validate which products have ran out when the first person bought all the stock.